indexer/cmd/indexer/cli.go

```diff
@@ -38,7 +38,7 @@ func runIndexer(ctx *cli.Context) error {
 	}
 	defer db.Close()
 
-	indexer, err := indexer.NewIndexer(log, db, cfg.Chain, cfg.RPCs, cfg.Metrics)
+	indexer, err := indexer.NewIndexer(log, db, cfg)
```

Comment:
Nothing should ever have access to the entire config object as was done intentionally in [https://github.com/ethereum-optimism/optimism/pull/6798](https://github.com/ethereum-optimism/optimism/pull/6798)

Reply:
this makes sense but with this addition, the indexer makes use of all parts of the config so I thought it'd be simpler to take in the entire thing config.

If config grows again & creates new things parts that the index service doesn't use, makes sense to take things in individually? wdyt?

Or I can create an `IndexerConfig` struct in this same file for brevity

Reply:
Gonna need to either do one of
1. `api.config.toml` `indexer.config.toml` thing
2. Same idea as 1 but keep `indexer.config.toml` as one config for convenience
3. Use seperate CLI apps for both
I don't think overloading a config option like configuring a server should be overloaded but instead be explicit. Going straight to one of these options works.

So all 3 of these options involve the config growing

Reply:
forsure, i'll just pass in each config struct explicitly